### PR TITLE
feat: support descent path operator

### PR DESCRIPTION
**Pull request by romange**

### CODE REVIEW

File: src/core/json/jsonpath_test.cc [hunk 1]
```diff
@@ -165,35 +165,88 @@ TEST_F(JsonPathTest, Path) {
   int called = 0;
 
   // Empty path
-  EvaluatePath(path, json, [&](const JsonType& val) { ++called; });
+  EvaluatePath(path, json, [&](optional<string_view>, const JsonType& val) { ++called; });
```
Comment (romange) [inline src/core/json/jsonpath_test.cc:168]: we pass key/value pair now. key is nullopt for array elements that do not have a key

Comment (BorysTheDev) [reply src/core/json/jsonpath_test.cc:168]: can it be empty string_view?

File: src/core/json/path.cc [hunk 2]
```diff
@@ -143,6 +149,15 @@ auto Dfs::Item::Init(const PathSegment& segment) -> AdvanceResult {
     }
 
     case SegmentType::DESCENT:
+      if (segment_step_ == 1) {
+        // first time, branching to return the same object but with the next segment,
+        // exploring the path of ignoring the DESCENT operator.
+        // Alsom, shift the state (segment_step) to bypass this branch next time.
```
Comment (BorysTheDev) [inline src/core/json/path.cc:155]: alsom)

**Review by BorysTheDev — APPROVED** [review]

File: src/core/json/jsonpath_test.cc (hunk 1 above)
Comment (romange) [reply src/core/json/jsonpath_test.cc:168]: Json key can be empty

**Pull request merged**